File: ios/LocalizedTimezoneName.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LocalizedTimezoneName, NSObject)

RCT_EXTERN_METHOD(getLocalizedTimeZoneName:(NSString *)timeZone locale:(NSString *)locale: completion:(RCTResponseSenderBlock *)callback)

@end

File: ios/LocalizedTimezoneName.swift
import Foundation

@objc(LocalizedTimezoneName)
class LocalizedTimezoneName: NSObject {

    @objc
    func getLocalizedTimeZoneName(timeZone: NSString, locale: NSString) -> NSString{
        let tz = NSTimeZone(name: timeZone as String)!
        return tz.localizedName(.standard, locale: Locale.init(identifier: locale as String)) as! NSString
    }
}
